src/main/java/org/prebid/server/bidder/connectad/ConnectadBidder.java

```diff
@@ -0,0 +1,199 @@
+                        .payload(outgoingRequest)
+                        .body(body)
+                        .build()),
+                errors);
+    }
+
+    private int secureFrom(Site site) {
+        final String page = site != null ? site.getPage() : null;
+        return StringUtils.isNotBlank(page) && page.startsWith(HTTPS_PREFIX) ? 1 : 0;
+    }
+
+    private ExtImpConnectAd parseImpExt(Imp imp) {
+        final ExtImpConnectAd extImpConnectAd;
+        try {
+            extImpConnectAd = mapper.mapper().convertValue(imp.getExt(), CONNECTAD_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(String.format("Impression id=%s, has invalid Ext", imp.getId()));
+        }
+
+        if (Objects.equals(extImpConnectAd.getSiteId(), NumberUtils.INTEGER_ZERO)) {
+            throw new PreBidException(String.format("Impression id=%s, has no siteId present", imp.getId()));
+        }
+        return extImpConnectAd;
+    }
+
+    private Imp updateImp(Imp imp, Integer secure, Integer siteId, BigDecimal bidFloor) {
+        final Imp.ImpBuilder updatedImp = imp.toBuilder().tagid(siteId.toString()).secure(secure);
+
+        if (bidFloor != null && bidFloor.compareTo(BigDecimal.ZERO) != NumberUtils.INTEGER_ZERO) {
+            updatedImp.bidfloor(bidFloor).bidfloorcur("USD");
+        }
+
+        final Banner banner = imp.getBanner();
+        if (banner == null) {
+            throw new PreBidException("We need a Banner Object in the request");
+        }
+
+        if (banner.getW() == null && banner.getH() == null) {
+            if (CollectionUtils.isEmpty(banner.getFormat())) {
+                throw new PreBidException("At least one size is required");
+            }
+            final Format format = banner.getFormat().get(0);
+            final List<Format> slicedFormatList = new ArrayList<>(banner.getFormat());
+
+            slicedFormatList.remove(0);
+            updatedImp.banner(banner.toBuilder().format(slicedFormatList).w(format.getW()).h(format.getH()).build());
+        }
+
+        return updatedImp.build();
+    }
+
+    final MultiMap resolveHeaders(Device device) {
+        final MultiMap headers = HttpUtil.headers();
+
+        if (device != null) {
+            updateHeadersWithStringValue(headers, "User-Agent", device.getUa());
+            updateHeadersWithStringValue(headers, "Accept-Language", device.getLanguage());
+            updateHeadersWithStringValue(headers, "X-Forwarded-For", device.getIp());
+            updateHeadersWithStringValue(headers, "X-Forwarded-For", device.getIpv6());
+            if (device.getDnt() != null) {
+                headers.add("DNT", device.getDnt().toString());
+            } else {
+                headers.add("DNT", "0");
+            }
+        }
+        return headers;
+    }
+
+    private void updateHeadersWithStringValue(MultiMap headers, String header, String value) {
```

Comment:
Just an advance. Headers can be `string` type only, so no need to specify this in name.
Rather, we can name it as `addHeaderIfValueIsNotBlank` or just `addHeader` or similar.

PS. make it static also.